packages/lesswrong/components/form-components/CoauthorsListEditor.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import { arrayMove } from 'react-sortable-hoc';
+import { registerComponent, Components } from '../../lib/vulcan-lib';
+import Checkbox from '@material-ui/core/Checkbox';
+import withUser from '../common/withUser';
+import { SortableList, shouldCancelStart } from './UsersListEditor';
+
+const coauthorsListEditorStyles = (theme: ThemeType): JssStyles => ({
+  root: {
+    display: 'flex',
+  },
+  checkbox: {
+    padding: '6px',
+  },
+  checkboxContainer: {
+    margin: '10px 0',
+    fontSize: '1.1rem',
+    fontWeight: 400,
+  },
+});
+
+const CoauthorsListEditor = ({ value, path, document, classes, label, currentUser, updateCurrentValues }: {
```

Comment:
The UserListEditor component is used in quite a lot of different places, so I created this as a separate component (which is really just a thin wrapper over the functionality that was already provided) to handle the new co-authors schema.